com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/dao/APIDAO.java

```diff
@@ -882,4 +882,56 @@ public static boolean inAPIQuotaDateRange(String serviceProvider,String applicat
 		return inQuotaDateRange;
 	}
 
+	public boolean isBlacklistedorWhitelistedNumber(String msisdn, String apiId,
+													String appId, String subscriberId,
+													String action) throws Exception {
+
+		Connection connection = null;
+		PreparedStatement preparedStatement = null;
+		ResultSet resultSet = null;
+		boolean isListed = false;
+
+		try {
+			StringBuilder queryBuilder = new StringBuilder();
+
+			queryBuilder.append("select exists ");
+			queryBuilder.append(" ( ");
+			queryBuilder.append("select 1 FROM ");
+			queryBuilder.append(DatabaseTables.BLACKLIST_WHITELIST_MSISDN + "  ");
+			queryBuilder.append(" where ");
+			queryBuilder.append(" (API_ID = '%' OR API_ID = ? )");
+			queryBuilder.append(" AND MSISDN = ? ");
+			queryBuilder.append(" AND (APP_ID = '%' OR APP_ID = ? )");
+			queryBuilder.append(" AND SERVICE_PROVIDER = ? ");
+			queryBuilder.append(" AND ACTION = ? ");
+			queryBuilder.append(" ) ");
+			queryBuilder.append(" AS result ");
+
+			connection = DbUtils
+					.getDbConnection(DataSourceNames.WSO2AM_STATS_DB);
+
+			preparedStatement = connection.prepareStatement(queryBuilder
+					.toString());
+			preparedStatement.setString(1, apiId);
+			preparedStatement.setString(2, msisdn);
+			preparedStatement.setString(3, appId);
+			preparedStatement.setString(4, subscriberId);
+			preparedStatement.setString(5, action);
+
+			resultSet = preparedStatement.executeQuery();
+
+			while (resultSet.next()) {
+				isListed = resultSet.getBoolean("result");
+			}
+
+		} catch (Exception ex) {
+			log.error("database operation error in API_ID :", ex);
+			throw ex;
+		} finally {
+			DbUtils.closeAllConnections(preparedStatement, connection,
```

Comment:
Use try with resource blocks. You can avoid tracking and closing held resources.